kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -172,8 +177,32 @@ impl<'tcx> MonoItemsCollector<'tcx> {
     /// instruction looking for the items that should be included in the compilation.
     fn reachable_items(&mut self) {
         while let Some(to_visit) = self.queue.pop() {
-            if !self.collected.contains(&to_visit) {
-                self.collected.insert(to_visit);
+            if let Entry::Vacant(e) = self.collected.entry(to_visit) {
```

Comment:
I think we should structure this code a bit better. Please create an issue and add a TODO to do that. Also, for now, can you at least move this code a separate function?